### DISCUSSION TITLE
VITE_SERVER_ENDPOINT setting for web server

### DISCUSSION THREAD

**Discussion by mprasil** [immich-app/immich#196] [Q&A]
I've noticed the web service is doing some requests browser-side and some requests are done by the service directly. The way I'm trying to set it up, these two requests should be using different URL. For browser-side requests, I'd like to use the ssl enabled proxy (so something like `https://immich.example.com`) and for the requests made directly between services I'd like to use the URL pointing to service on the docker network directly (so working with something similar to the [provided docker compose file](https://github.com/alextran1502/immich/blob/main/docker/docker-compose.yml) this would be `http://immich-server:3000`)

However as far as I have found, only single `VITE_SERVER_ENDPOINT` setting is available. Is there any way to configure this separately or perhaps to make the web service to only do browser-side or server-side requests only? I'm actually curious why are the requests done this way, I'm not familiar with the framework used.

(great service BTW, I'm very impressed)

**Comment by alextran1502** [comment]
Hello,

Thank you for the kind words. SvelteKit can be seen as Next.js or Nuxt.js for React and Vue, so it has the ability to perform SSR, so most of the requests for authentication are done on the server-side of the framework, and other requests are a combination of both client-side and server-side depends on the needs of fast page load speed. 

During the website build process, Vite will inject the `VITE_SERVER_ENDPOINT` content into the web app.

Since the browser side doesn't have visibility into the internal docker-compose network, I don't think it is possible to make a request to let's say `http//immich-server:3000` 

I believe you can still use SSL proxy by pointing the endpoint to your internal service on port 2283. This is a relevant comment from another user https://github.com/alextran1502/immich/issues/170#issuecomment-1138133292 that you might find helpful

I hope this answers your question.

**Reply by mprasil** [reply]
> Since the browser side doesn't have visibility into the internal docker-compose network, I don't think it is possible to make a request to let's say http//immich-server:3000

This is what I was wondering - would it be possible to have separate setting for SSR and for browser-side request? Something like `VITE_SERVER_SSR_ENDPOINT` and `VITE_SERVER_PUBLIC_ENDPOINT`. Something that would let me inform the web app where those two endpoints can be reached instead of one common variable.
